src/zc/lockfile/README.txt

```diff
@@ -68,3 +68,23 @@ If you now inspected the lock file, you would see e.g.:
     $ cat lock
      123;myhostname
 
+Context Managers
+================
+
+This module also implements two Context Managers, in addition to the primitives
+discussed above. The first just gets and releases a lock file, with a timeout.
+
+    >>> rough_seconds = 0.7
+    >>> with zc.lockfile.LockedContext('description', timeout=rough_seconds):
+    ...    pass
+
+The lock will always be released when you exit this scope. If timeout is not
+given, it is assumed you wish to wait forever. It should be noted that this
+Context Manager will always add '.lock' to the end of any filename you give it.
```

Comment:
This seems wrong to me. It seems more flexible, and more likely to gain adoption among existing uses of `zc.lockfile` if this safer, more convenient version of `LockFile` just locked the file that you give it.

Then the `open_exclusive` manager would be implemented by *composition* with this context manager, not by subclassing. 

Something like:

```python
from contextlib import contextmanager
@contextmanager
def open_exclusive(self, filename, lock_file_name=None, **open_args):
        lock_file_name = lock_file_name or filename + os.path.extsep + 'lock'
        with locked_file(lock_file_name):
            with open(filename, **open_args) as f:
                   yield f
```